url7895f5cfa7_edm92_defaultlogic_tgz-pJ8-be_fnord_DefaultLogic_AppJ8/src/be/fnord/util/logic/defaultLogic/RuleSet.java

```diff
@@ -0,0 +1,348 @@
+				// a.e.println("Trying: " + currentExtension.getFormula());
+				// Test if prerequisite is fired
+				WFF preq = new WFF(d.getPrerequisite());
+				WFF just = new WFF(d.getJustificatoin());
+				WFF cons = new WFF(d.getConsequence());
+				boolean prereqisite_fired = assertion.eval(preq);
+				// System.out.println(assertion.getFormula() + "  |=  " +
+				// preq.getFormula() + " ===" + prereqisite_fired);
+
+				if (prereqisite_fired) {
+					boolean just_consistent = assertion.isConsistent(just
+							.getFormula());
+					if (just_consistent) {
+						// Test if consequence is entailed by extension
+						boolean test_consequence_entailed_by_extension = currentExtension
+								.entails(cons);
+						// System.out.println(currentExtension.getFormula() +
+						// " |= " + d.getConsequence() + " = " +
+						// test_consequence_entailed_by_extension );
+
+						if (!test_consequence_entailed_by_extension) {
+							overall = false;
+						}
+						_consequences.add(currentExtension.getFormula());
+					}
+				} else {
+
+					//
+					//
+					// if(!preqCon.isConsistent(currentExtension.getFormula())){
+					// System.out.println("false");
+					// }
+				}
+			}
+			// System.out.println("_consequences:" + _consequences);
+			// System.out.println("~~-------------------------------------------------------");
+			if (overall) {
+				// Loop through the rules again looking for inconsistencies
+				for (DefaultRule d : rules) {
+					WFF preq = new WFF(d.getPrerequisite());
+					// WFF just = new WFF(d.getJustificatoin());
+					WFF cons = new WFF(d.getConsequence());
+					boolean prereqisite_fired = assertion.entails(preq);
+					// boolean just_consistent =
+					// assertion.isConsistent(just.getFormula());
+
+					if (!prereqisite_fired) {
+						WFF preqCon = new WFF("(" + d.getPrerequisite()
+								+ ") & (" + d.getJustificatoin() + ")");
+						boolean test_consequence_entailed_by_extension = currentExtension
+								.entails(cons);
+						if (test_consequence_entailed_by_extension) {
+							if (!preqCon.isConsistent(currentExtension
+									.getFormula())) {
+								overall = false;
+							}
+						}
+					}
+				}
+				if (!overall) {
+					break;
+				}
+				// We create a deductive closure of the extension and all of the
+				// consequences
+				String talliedCons = "";
+				for (String c : _consequences) {
+					talliedCons = talliedCons + " " + a.e.AND + " " + c;
+				}
+				if (talliedCons.length() > 0) {
+					talliedCons = talliedCons.substring(
+							(" " + a.e.AND + " ").length(),
+							talliedCons.length());
+				}
+				WFF wffCons = new WFF(talliedCons);
+
+				String extString = currentExtension.getClosure().trim();
+				String conString = wffCons.getClosure().trim();
+				// System.out.println("ext =" + extString);
+				if (extString.compareTo(conString) == 0) {
+					_extensions.add(possExtension);
+				}
+			}
+		}
+
+		return _extensions;
+	}
+
+	public LinkedList<String> generateExtensions(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.